lib/core/src/Cardano/Wallet/Api/Server.hs

```
@@ -1778,13 +1824,27 @@ rndStateChange ctx (ApiT wid) pwd =
 -- | Makes an 'ApiCoinSelection' from the given 'UnsignedTx'.
 mkApiCoinSelection
     :: forall n. ()
-    => UnsignedTx (TxIn, TxOut, NonEmpty DerivationIndex)
+    => Maybe (DelegationAction, [DerivationIndex])
+    -> UnsignedTx (TxIn, TxOut, NonEmpty DerivationIndex)
     -> ApiCoinSelection n
-mkApiCoinSelection (UnsignedTx inputs outputs) =
+mkApiCoinSelection mcerts (UnsignedTx inputs outputs) =
     ApiCoinSelection
         (mkApiCoinSelectionInput <$> inputs)
         (mkAddressAmount <$> outputs)
+        (fmap (uncurry mkCertificates) mcerts)
   where
+    mkCertificates :: DelegationAction -> [DerivationIndex] -> [Api.ApiCertificate]
+    mkCertificates action (s:sx) =
+        let apiStakePath = ApiT <$> (s:|sx)
+        in case action of
+            Join pid -> [Api.JoinPool apiStakePath (ApiT pid)]
+            RegisterKeyAndJoin pid ->
+                [ Api.RegisterRewardAccount apiStakePath
+                , Api.JoinPool apiStakePath (ApiT pid)
+                ]
+            Quit-> [Api.QuitPool apiStakePath]
+    mkCertificates _ _ = []
```

This is silently swallowing an error. We should never ever have an empty derivation path here. Having one is clearly a sign that something went really wrong in the internal logic and should be loudly visible / should crash the application.